[DOCS] Rewrite `terms_set` query

jrodewig opened the pull request:
## Changes
* Rewrites description of `terms_set` query
* Adds example request section
* Adds parameters sections
* Adds notes section

This is part of #40977, an effort to standardize documentation for query types.

## Before
<details>
 <summary>Before image</summary>
<img width="774" alt="Before image" src="https://user-images.githubusercontent.com/40268737/59195661-85d29a00-8b5a-11e9-9a77-bc1e1c516036.png">
</details>


## After
<details>
 <summary>After image</summary>
<img width="774" alt="Before image" src="https://user-images.githubusercontent.com/40268737/60353493-20730b80-9998-11e9-9555-b6a16ace2e45.png">
</details>

jrodewig committed:
[DOCS] Rewrite `terms_set` query

elasticmachine commented:
Pinging @elastic/es-search

cbuescher commented on docs/reference/query-dsl/terms-set-query.asciidoc:
```diff
@@ -1,121 +1,220 @@
 
-An example that always limits the number of required terms to match to never
-become larger than the number of terms specified:
+`boost`::
```
Like in a previous PR, maybe we could mention the "boost" parameter on all queries for completeness but link to a common paragraph for the detailed discussion.

cbuescher reviewed (commented):
Thanks again @jrodewig, I didn't know this query very well and it was a pleasure reading the rewrite. Left one suggestions regarding "boost" again, also I wasn't sure about real-life use cases for this type of query so I did a bit of digging when/why this was added and found https://github.com/elastic/elasticsearch/issues/26915. There are two more or less intuitive examples where this pretty specialized query might be useful in https://github.com/elastic/elasticsearch/issues/26915#issuecomment-335100412. I don't know how far you want to take details in these reference docs but maybe there is a way of including some of these thoughts succinctly? Just an idea, maybe that makes the docs to complicated so I leave it up to you.

jrodewig committed:
Remove generic boost parm documentation

jrodewig committed:
Remove errant delimiter

jrodewig committed:
Add use case examples

jrodewig committed:
Reword several instances of "minimum number"

jrodewig commented:
Thanks for the review and praise @cbuescher. 

I've updated the example and included the listed use cases with a33ee6e. Thanks for pointing out #26915. I feel like the examples are much more practical now.

I also removed the `boost` parameter documentation for now with 75e12a4.

cbuescher reviewed (approved):
Awesome, LGTM
